docs/angular/rxobject.md

```diff
@@ -0,0 +1,416 @@
+  // Implementation here
+}
+```
+
+> **NOTE:** We use the keyword `export` on this class so that other TypeScript classes in our file can access this.
+
+Notice that we also make this class **`abstract`**. The core motivation behind in the `RxObject<T>` class is that we *do not instantiate it directly*. Instead, we want to have *subclasses* ***extend*** this base class for related usecases. For example, if we wanted to dynamically store data for organizations, we may want to create a subclass `RxOrganizationsList` that *extends* `RxObject<Organization[]>`. We will utilize this in the future, but for now, we are just working on the conceptual `abstract` class.
+
+From here, we know we need to successfully implement the Observer design pattern. We know that the Angular Components are going to be the *observers*, which means the only thing missing here is the *subject*. We want our `RxObject<T>` class to contain our subject.
+
+You can recall that, after examining the various options for our RxJS subject object, we decided that `ReplaySubject<T>` is the best for our use case. This is because the `ReplaySubject<T>` re-emits the latest value that is has to *new subscribers*. We will use a `ReplaySubject<T>` object to serve as the main *subject* to satisfy the requirements of the Observer design pattern.
+
+So, we need to add a `ReplaySubject<T>` to our `RxObject<T>` class. We want our `RxObject<T>` class to *abstract* away this functionality from the Components that use it, so we can make our `ReplaySubject<T>` a **private** field. We do not want to directly expose this subject to Components - rather, the subject will be used internally within the `RxObject<T>` object to manage the data stream. Let's add it to the class:
+
+```ts
+export abstract class RxObject<T> {
+
+  /**
+  * The Subject provides a "multicast" mechanism for publishing changes to observers.
+  * We choose a ReplaySubject with a value of 1 such that every new observer is guaranteed
+  * to receive the latest value immediately upon observation, once an initial value has
+  * been set.
+  */
+  private subject: Subject<T> = new ReplaySubject(1);
+
+}
+```
+
+As you can see, we *initialize* our `subject` field using a new `ReplaySubject` with a buffer size of `1` so that it stores just the most up-to-date value.
+
+So, if this field is private, then how do we want our Components to interact with our object? We want the Components to interact with the object by *subscribing to a simple `Observable<T>`*. The motivation behind this is so that `subject` manages the data stream and ensures that everything is working correctly behind the scenes, and that our Components do not need to worry about this concern and can work with the more general `Observable` type everywhere else. In addition, it enables the use of the `| async` pipe in HTML. Since we want our Components to access this field, we can make it public. Let's add the observable, made from our `subject`, into our object:
```

Comment:
Here's another place where I wonder if the notion of subject being read/write and managed internally and its observable being read-only is important. We're trying to keep the logic for notification encapsulated in the `RxObject` subclasses (through the helper method you're about to reveal).